### PR TITLE
Feature/add .editorconfig

### LINKED ISSUE #1: Add .editorconfig file to prevent code formatting issues

### PR DESCRIPTION
**Pull request by mrister**
Add .editorconfig. Closes #1 
Update .gitignore.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*]
```
Comment (jkernech) [inline .editorconfig]: Why did you repeat the wildcard section ?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+indent_style = space
+indent_size =
```
Comment (jkernech) [inline .editorconfig]: By default, `indent_size` should be at 2.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+[Makefile]
+indent_style = tab
+
+[*.md]
+indent_style = space
```
Comment (jkernech) [inline .editorconfig]: I think we can remove this line since this is the default behaviour.

File: .editorconfig (hunk 1 above)
Comment (mrister) [inline .editorconfig]: mistake

File: .editorconfig (hunk 2 above)
Comment (mrister) [inline .editorconfig]: yup, sorry bout that, update coming.

**Comment by jkernech** [comment]
LGTM !

**Comment by PuKoren** [comment]
LGTM

**Pull request merged**